tools/shared.py

```diff
@@ -127,6 +127,17 @@ def run_process(cmd, check=True, input=None, *args, **kw):
   sys.stderr.flush()
```

Comment:
Perhaps we could add an assertion here. e.g `assert len(shlex_join(cmd)) <= MAX_CMDLINE_LEN`.

Reply:
We'll get a failure from `subprocess.run()` in such a case, so I don't see the point in having to define a maximum command line length for each supported platform and adding an assertion here.

Reply:
Fair enough.  As long as the OS message to clear enough.

Reply:
At least on Windows, the returned error is `ERROR_FILENAME_EXCED_RANGE`, which results in the following Python exception: `FileNotFoundError: [WinError 206] The filename or extension is too long`.